ops/testing.py

```diff
@@ -446,35 +444,37 @@ def _next_relation_id(self):
         self._relation_id_counter += 1
         return rel_id
 
-    def add_storage(self, storage_name: str, count: int = 1) -> typing.List[str]:
-        """Declare a new storage device attached to this unit.
+    def add_storage(self, storage_name: str, count: int = 1,
+                    *, attach: bool = False) -> typing.List[str]:
+        """Create a new storage device and attach it to this unit.
 
         To have repeatable tests, each device will be initialized with
-        location set to /<storage_name>N, where N is the counter and
-        will be a number from [0,total_num_disks-1]
+        location set to /[tmpdir]/<storage_name>N, where N is the counter and
```

Comment:
do we want control over the tempdir per-storage?

Reply:
or do we not actually simulate reading/writing to the storage?

Reply:
The tmpdir is handled automatically.  And can be used for reading+writing.

Reply:
The tmpdir is obviously a harness specific thing.  I was just making sure that the docstring didn't lie.